ORM/DataObject.php

```diff
@@ -270,88 +283,7 @@ class DataObject extends ViewableData implements DataObjectInterface, i18nEntity
 	 * @return DataObjectSchema
 	 */
 	public static function getSchema() {
-		return Injector::inst()->get('SilverStripe\ORM\DataObjectSchema');
-	}
-
-	/**
-	 * Return the complete map of fields to specification on this object, including fixed_fields.
-	 * "ID" will be included on every table.
-	 *
-	 * Composite DB field specifications are returned by reference if necessary, but not in the return
-	 * array.
-	 *
-	 * Can be called directly on an object. E.g. Member::database_fields()
-	 *
-	 * @param string $class Class name to query from
-	 * @return array Map of fieldname to specification, similiar to {@link DataObject::$db}.
-	 */
-	public static function database_fields($class = null) {
-		if(empty($class)) {
-			$class = get_called_class();
-		}
-		return static::getSchema()->databaseFields($class);
-	}
-
-	/**
-	 * Get all database columns explicitly defined on a class in {@link DataObject::$db}
-	 * and {@link DataObject::$has_one}. Resolves instances of {@link DBComposite}
-	 * into the actual database fields, rather than the name of the field which
-	 * might not equate a database column.
-	 *
-	 * Does not include "base fields" like "ID", "ClassName", "Created", "LastEdited",
-	 * see {@link database_fields()}.
-	 *
-	 * Can be called directly on an object. E.g. Member::custom_database_fields()
-	 *
-	 * @uses DBComposite->compositeDatabaseFields()
-	 *
-	 * @param string $class Class name to query from
-	 * @return array Map of fieldname to specification, similiar to {@link DataObject::$db}.
-	 */
-	public static function custom_database_fields($class = null) {
-		if(empty($class)) {
-			$class = get_called_class();
-		}
-
-		// Remove fixed fields. This assumes that NO fixed_fields are composite
-		$fields = static::getSchema()->databaseFields($class);
-		$fields = array_diff_key($fields, self::config()->fixed_fields);
-		return $fields;
-	}
-
-	/**
-	 * Returns the field class if the given db field on the class is a composite field.
-	 * Will check all applicable ancestor classes and aggregate results.
-	 *
-	 * @param string $class Class to check
-	 * @param string $name Field to check
-	 * @param boolean $aggregated True if parent classes should be checked, or false to limit to this class
-	 * @return string|false Class spec name of composite field if it exists, or false if not
-	 */
-	public static function is_composite_field($class, $name, $aggregated = true) {
```

Comment:
The removal of all of these static methods to a non-static DataObjectSchema service means we can now DI our schema. :)